### ISSUE TITLE
Objects with readonly fields can assigned to objects without having readonly modifier

### ISSUE THREAD

**Issue by cevek** [Microsoft/TypeScript#16080]
**TypeScript Version:**  2.4.0-dev.20170519

**Code**

```ts
interface FooReadonly {
    readonly foo: number;
}

interface Foo {
    foo: number;
}
let y:FooReadonly = {foo: 1};
let x:Foo = y;
x.foo = 2;
```

**Expected behavior:**
Should throw an error

**Actual behavior:**
No error

**Comment by andy-ms** [comment]
Duplicate of #13347

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
